pennylane/qnodes/base.py

```diff
@@ -740,11 +721,13 @@ def evaluate(self, args, kwargs):
 
         self.device.reset()
 
+        temp = self.properties.get('use_native_type', False)
         if isinstance(self.device, qml.QubitDevice):
```

Comment:
@josh146 Not really part of this PR, but does it make sense for the base class to know about its child classes and implement specific logic for them?

Reply:
Hmm, I would say no.

Reply:
Oh, do you mean the `if isinstance(self.device, qml.QubitDevice):` line?

In this case, the QNode is just checking which device it is about to evaluate (whether the old Device or a new QubitDevice), as they have different execute signatures.

This is a temporary change, and once the CVDevice is implemented, and all plugins ported to QubitDevice, can be removed.

Reply:
```suggestion
        if isinstance(self.device, qml.QubitDevice):
            #TODO: remove this if statement once all devices are ported to the QubitDevice API
```